### PR TITLE
Cleanup and improvements

### LINKED ISSUE #169: Support different timetables for different campuses
### Is your feature request related to a problem? Please describe.
Currently we only have one timetable for both Vellore and Chennai campuses. This works well if timetable followed by both campuses are same. But in case the timetable is different for both the campuses we don't have a way to support both of them.

### Describe the solution you'd like
Add support for different timetables for different campuses. This will require good efforts in terms of refactoring existing code.
Must ideal refactoring is moving to react or preact but it's lot of efforts. It'll also require re-writing all of jquery code.
Easiest approach I can think of is to write a helper function which takes a schema of timetable and renders it accordingly. We can call this function with correct schema when changing the campus and re-render the timetable. 

### Describe alternatives you've considered
In past, similar problem was solved by deploying different code on a different route. I don't think it's a good approach.

### Additional context
#168

### PR DESCRIPTION
**Pull request by therealsujitk**
Hi @vatz88 I made quite a few changes in this PR but don't worry you don't have to look at the code, I'll list down the things I've changed below.

- Moved SCSS rules into separate files and removed rules no longer in use. In the past you were hesitant to use custom CSS because it might not have cross browser support, so I've moved as much as I could to bootstrap and now there's barely any custom CSS (I'm surprised too).
- Moved JS functions into separate files, removed redundant and deprecated code. You were concerned with how I was storing a bunch of data and functions in the window object, I googled about that and looks like you were right, while requiring a package in nodejs as long as it's in the same directory it always gives the same instance. However I think Bootstrap 4 requires jQuery to be a window object because of how it access' it, one of the packages we use requires BS4 for now, however I might be able to resolve this once they release a stable version with BS5 support.
- I've added separate last update fields for vellore and chennai for obvious reasons and also moved the last updated semester to a more viewable location. This value changes when the semester is changed.
- I've changed the ascending and descending arrows to use FontAwesome icons instead making it more UI friendly. Also, I changed ascending to an up-arrow and descending to a down-arrow (it was reversed), after someone mentioned it in the past I looked around and found people saying the pointed part is the smaller part. If you want me to change it back let me know.
- And finally the x-axis padding in the timetable and course list, you wanted them to be full width but it didn't look very good on larger devices, so I've added a breakpoint to use full width on smaller devices only (personally I think it looks better if smaller devices had a small amount padding as well but this works too).

If you have any suggestions please let me know, **I won't be merging this PR immediately** (Edit: It looks like I don't have permission to merge my own PRs). There's still a tiny amount of cleaning left. I'll merge it after I resolve #169. I did this cleanup first so that everything would be more organised for the major changes I'll have to do to support different timetables.

This PR closes most of #193.

### CODE REVIEW

**Comment by therealsujitk** [comment]
Also, I'd like to increase the required node version for this application to 14 or some other higher LTS version. Node.js 12 will be reaching [end of life](https://endoflife.date/nodejs) soon.

File: src/scss/timetable.scss [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        justify-content: center !important;
+    }
+}
+
+@media screen and (max-width: 1000px) {
+    /*
+        Hiding the text in the option buttons if the screen
+        width is not big enough
+    */
+    button span {
```
Comment (vatz88) [inline src/scss/timetable.scss:149]: `button#option-buttons span` ?
To be more specific

**Comment by vatz88** [comment]
Sounds good to me!

**Comment by vatz88** [comment]
Should the download course list be a separate button somewhere in course panel?

image

"Download timetable" is not very obvious to have two options.

File: src/scss/timetable.scss (hunk 1 above)
Comment (therealsujitk) [reply src/scss/timetable.scss:149]: Sure, I'll add this.

**Comment by vatz88** [comment]
> Should the download course list be a separate button somewhere in course panel?
> 
> image
> 
> "Download timetable" is not very obvious to have two options.

Ah my bad, i confused it with https://github.com/vatz88/FFCSonTheGo/issues/209

**Review by vatz88 — APPROVED** [review]

**Comment by therealsujitk** [comment]
> "Download timetable" is not very obvious to have two options.

I can't think of calling it anything else. I had previously planned to add a third option in that modal where users can download a PDF containing both of them. (Might add it in the future, the code is somewhere in my fork)

> Should the download course list be a separate button somewhere in course panel?

I'll try moving things around and see if something looks good.

**Comment by therealsujitk** [comment]
Also, could you unlink the issue this PR is linked to, It got linked automatically when when I mentioned it. It should be linked to #193.

**Comment by vatz88** [comment]
GitHub isn't letting me 🤦‍♂️
Also, you're added as contributor to this repo so if I can you can as well, or we both can't

**Comment by therealsujitk** [comment]
> GitHub isn't letting me 🤦‍♂️

Oh, I thought I didn't have permission to do it. Must be a GitHub bug.

**Pull request merged**